snuba/migrations/snuba_migrations/functions/0002_add_thread_id.py

```diff
@@ -0,0 +1,180 @@
+from typing import List, Sequence
+
+from snuba.clickhouse.columns import (
+    UUID,
+    AggregateFunction,
+    Column,
+    DateTime,
+    Float,
+    String,
+    Tuple,
+    UInt,
+)
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+
+columns: List[Column[Modifiers]] = [
+    Column("project_id", UInt(64)),
+    Column("transaction_name", String()),
+    Column("timestamp", DateTime()),
+    Column("depth", UInt(32)),
+    Column("parent_fingerprint", UInt(64)),
+    Column("fingerprint", UInt(64)),
+    Column("name", String()),
+    Column("package", String()),
+    Column("path", String()),
+    Column("is_application", UInt(8)),
+    Column("platform", String(Modifiers(low_cardinality=True))),
+    Column("environment", String(Modifiers(nullable=True, low_cardinality=True))),
+    Column("release", String(Modifiers(nullable=True, low_cardinality=True))),
+    Column("os_name", String(Modifiers(low_cardinality=True))),
+    Column("os_version", String(Modifiers(low_cardinality=True))),
+    Column("retention_days", UInt(16)),
+    Column("count", AggregateFunction("count", [Float(64)])),
+    Column(
+        "percentiles",
+        AggregateFunction("quantiles(0.5, 0.75, 0.9, 0.95, 0.99)", [Float(64)]),
+    ),
+    Column("min", AggregateFunction("min", [Float(64)])),
+    Column("max", AggregateFunction("max", [Float(64)])),
+    Column("avg", AggregateFunction("avg", [Float(64)])),
+    Column("sum", AggregateFunction("sum", [Float(64)])),
+    Column(
+        "worst", AggregateFunction("argMax", [Tuple((UUID(), UInt(64))), Float(64)])
+    ),
+    Column(
+        "examples", AggregateFunction("groupUniqArray(5)", [Tuple((UUID(), UInt(64)))])
+    ),
+]
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+    index_granularity = "2048"
+    storage_set = StorageSetKey.FUNCTIONS
+
+    data_granularity = 60 * 60  # 1 hour buckets
+
+    local_raw_table = "functions_raw_local"
+    dist_raw_table = "functions_raw_dist"
+
+    local_materialized_table = "functions_mv_local"
+    dist_materialized_table = "functions_mv_dist"
+
+    local_view_table = "functions_local"
+
+    def forwards_local(self) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.AddColumn(
+                storage_set=self.storage_set,
+                table_name=self.local_raw_table,
+                column=Column("thread_id", UInt(64)),
+                after="profile_id",
+            ),
+            operations.DropTable(
+                storage_set=self.storage_set, table_name=self.local_materialized_table
+            ),
+            operations.DropTable(
+                storage_set=self.storage_set, table_name=self.local_view_table
+            ),
```

Comment:
Not sure if there's a better way to do this but dropping the existing data is not a problem right now.